aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/SubscriptionAuthorizer.java

```diff
@@ -199,8 +205,28 @@ private static String getDataJson(Subscription subscription) {
         }
     }
 
-    private static String getApiUrl(AWSConfiguration awsConfiguration) throws JSONException {
-        return awsConfiguration.optJsonObject("AppSync").getString("ApiUrl");
+    private String getRegion() throws JSONException {
+         return mRegionStr != null
+                 ? mRegionStr
```

Comment:
`mRegionStr` is the AppSync region. The region used for `credentialProvider` construction should always be the identity pool region. Theoretically, the two can be different and so I think we should just read it from the configuration json. In that case may we get rid of mRegionStr altogether since its not used anywhere else?

Reply:
Okay that works. Will update soon.

Reply:
Done 👍